src/socket.rs

```diff
@@ -249,6 +274,7 @@ where
         while let Some(result) = self.stream.next().await {
             let result = result.map(M::into);
             tracing::trace!("received message: {:?}", result);
+            *self.last_alive.lock().await = Instant::now();
```

Comment:
AFAIK there's no standard defined if we should handle the keep alive timer after Pong or after any message.
so yeah it makes sense to use keep-alive there.